bpfd/src/oci_utils/image_manager.rs

```diff
@@ -67,37 +78,118 @@ impl BytecodeImage {
     pub(crate) fn get_pull_policy(self) -> ImagePullPolicy {
         self.image_pull_policy
     }
+}
+
+impl From<bpfd_api::v1::BytecodeImage> for BytecodeImage {
+    fn from(value: bpfd_api::v1::BytecodeImage) -> Self {
+        BytecodeImage::new(
+            value.url,
+            value.image_pull_policy,
+            match value.username.as_ref() {
+                "" => None,
+                u => Some(u.to_string()),
+            },
+            match value.password.as_ref() {
+                "" => None,
+                p => Some(p.to_string()),
+            },
+        )
+    }
+}
+
+pub(crate) struct ImageManager {
+    base_dir: PathBuf,
+    client: Client,
+    rx: Receiver<Command>,
+}
+
+/// Provided by the requester and used by the manager task to send
+/// the command response back to the requester.
+type Responder<T> = oneshot::Sender<T>;
+
+#[derive(Debug)]
+pub(crate) enum Command {
+    Pull {
+        image: String,
+        pull_policy: ImagePullPolicy,
+        username: Option<String>,
+        password: Option<String>,
+        resp: Responder<Result<(String, String), ImageError>>,
+    },
+    GetBytecode {
+        path: String,
+        resp: Responder<Result<Vec<u8>, anyhow::Error>>,
```

Comment:
Also after reading more... Should we just return the PATH to the bytecode here on disk rather than the raw bytes? Then in the core BpfManager we can always just use `crate::utils::read(l)` on the file.  It'd also mean less data going over the channel

Reply:
When I started this work I wanted to collapse the two API commands Pull and GetBytecode into one.
We can’t call `crate::utils::read` the path that comes back from `Pull` since its a directory in our image store.
The `GetBytecode` API does all the fancy things from that parh - fetch manifest, find layers, ungzip them and return the bytes etc…

ISTR we discussed having a cache of inflated layers on disk, but decided against it because without some form of persistent database we wouldn’t know if any of those layers had been tampered with - i.e a malicious actor replaces the bytecode while bpfd is running and/or restarting.

My understanding of tokio channels is that passing Vec’s is cheap, because a Vec is just a pointer (the data is on the heap). Since we're copying the pointer, not the heap data, this should be fine 🤔 

In other words, this is the least-expensive short-term option 😆

Reply:
> My understanding of tokio channels is that passing Vec’s is cheap, because a Vec is just a pointer (the data is on the heap). Since we're copying the pointer, not the heap data, this should be fine 

I like it sounds good to me